Library/OptimizelyClientType.swift

```diff
@@ -62,10 +63,63 @@ extension OptimizelyClientType {
 
     return variant
   }
+
+  /* Returns all experiments the app knows about */
+
+  public func allExperiments() -> [String] {
+    return OptimizelyExperiment.Key.allCases.map { $0.rawValue }
+  }
 }
 
 // MARK: - Tracking Properties
 
+public func optimizelyProperties(environment: Environment? = AppEnvironment.current) -> [String: Any]? {
+  guard let env = environment, let optimizelyClient = env.optimizelyClient else {
+    return nil
+  }
+
+  let environmentType = env.environmentType
+  let userId = deviceIdentifier(uuid: UUID())
+  let attributes = optimizelyUserAttributes()
+
+  var sdkKey: String
+
+  switch environmentType {
+  case .production:
+    sdkKey = Secrets.OptimizelySDKKey.production
+  case .staging:
+    sdkKey = Secrets.OptimizelySDKKey.staging
+  case .development, .local:
+    sdkKey = Secrets.OptimizelySDKKey.development
+  }
+
+  let allExperiments = optimizelyClient.allExperiments().map { experimentKey -> [String: String] in
+    do {
```

Comment:
Given that we aren't handling the exception, I think we only care about the value being `nil` so we could get away without repeating the dictionary keys:
```swift
let allExperiments = optimizelyClient.allExperiments().map { experimentKey -> [String: String] in
  let variation = try? optimizelyClient.getVariationKey(
    experimentKey: experimentKey,
    userId: userId,
    attributes: attributes
  )

  return [
    "optimizely_experiment_slug": experimentKey,
    "optimizely_variant_id": variation ?? "unknown"
  ]
}
```